src/utilities/resolveSchemaCoordinate.ts

```
@@ -0,0 +1,190 @@
+    }
+  | {
+      readonly kind: 'FieldArgument';
+      readonly type: GraphQLNamedType;
+      readonly field: GraphQLField<unknown, unknown>;
+      readonly fieldArgument: GraphQLArgument;
+    }
+  | {
+      readonly kind: 'Directive';
+      readonly directive: GraphQLDirective;
+    }
+  | {
+      readonly kind: 'DirectiveArgument';
+      readonly directive: GraphQLDirective;
+      readonly directiveArgument: GraphQLArgument;
+    };
+
+/**
+ * A schema coordinate is resolved in the context of a GraphQL schema to
+ * uniquely identifies a schema element. It returns undefined if the schema
+ * coordinate does not resolve to a schema element.
+ *
+ * https://spec.graphql.org/draft/#sec-Schema-Coordinates.Semantics
+ */
+export function resolveSchemaCoordinate(
+  schema: GraphQLSchema,
+  schemaCoordinate: string | Source,
+): ResolvedSchemaElement | undefined {
+  return resolveASTSchemaCoordinate(
+    schema,
+    parseSchemaCoordinate(schemaCoordinate),
+  );
+}
+
+/**
+ * Resolves schema coordinate from a parsed SchemaCoordinate node.
+ */
+export function resolveASTSchemaCoordinate(
+  schema: GraphQLSchema,
+  schemaCoordinate: SchemaCoordinateNode,
+): ResolvedSchemaElement | undefined {
+  const { ofDirective, name, memberName, argumentName } = schemaCoordinate;
+  if (ofDirective) {
+    // SchemaCoordinate :
+    //   - @ Name
+    //   - @ Name ( Name : )
+    // Let {directiveName} be the value of the first {Name}.
+    // Let {directive} be the directive in the {schema} named {directiveName}.
+    const directive = schema.getDirective(name.value);
+    if (!argumentName) {
+      // SchemaCoordinate : @ Name
+      // Return the directive in the {schema} named {directiveName}.
+      if (!directive) {
+        return;
+      }
+      return { kind: 'Directive', directive };
+    }
+
+    // SchemaCoordinate : @ Name ( Name : )
+    // Assert {directive} must exist.
+    if (!directive) {
+      return;
```

Should assertions throw instead of returning `undefined`?